### ISSUE TITLE
[iOS] Version 4.3.1 doesn't complie in debug mode for simulator

### ISSUE THREAD

**Issue by lorenc-tomasz** [sentiance/react-native-sentiance#152]
From version 4.3.0 SDK cannot be compiled and run on x86 simulator.

**Comment by lorenc-tomasz** [comment]
Example error when using 4.3.x:

`Undefined symbols for architecture x86_64:
  "boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::do_assign(char const*, char const*, unsigned int)", referenced from:
      boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::assign(char const*, char const*, unsigned int) in libFlipper-Folly.a(Benchmark.o)
      boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::assign(char const*, char const*, unsigned int) in libFlipper-Folly.a(Uri.o)
  "boost::re_detail_106300::perl_matcher<char const*, std::__1::allocator<boost::sub_match<char const*> >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::construct_init(boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags)", referenced from:
      boost::re_detail_106300::perl_matcher<char const*, std::__1::allocator<boost::sub_match<char const*> >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::perl_matcher(char const*, char const*, boost::match_results<char const*, std::__1::allocator<boost::sub_match<char const*> > >&, boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags, char const*) in libFlipper-Folly.a(Uri.o)
  "boost::re_detail_106300::perl_matcher<char const*, std::__1::allocator<boost::sub_match<char const*> >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::find()", referenced from:
      bool boost::regex_search<char const*, std::__1::allocator<boost::sub_match<char const*> >, char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >(char const*, char const*, boost::match_results<char const*, std::__1::allocator<boost::sub_match<char const*> > >&, boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags, char const*) in libFlipper-Folly.a(Uri.o)
  "boost::re_detail_106300::perl_matcher<char const*, std::__1::allocator<boost::sub_match<char const*> >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::match()", referenced from:
      bool boost::regex_match<char const*, std::__1::allocator<boost::sub_match<char const*> >, char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >(char const*, char const*, boost::match_results<char const*, std::__1::allocator<boost::sub_match<char const*> > >&, boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags) in libFlipper-Folly.a(Uri.o)
  "boost::re_detail_106300::get_mem_block()", referenced from:
      boost::re_detail_106300::perl_matcher<std::__1::__wrap_iter<char const*>, std::__1::allocator<boost::sub_match<std::__1::__wrap_iter<char const*> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::extend_stack() in libFlipper-Folly.a(Benchmark.o)
      boost::re_detail_106300::save_state_init::save_state_init(boost::re_detail_106300::saved_state**, boost::re_detail_106300::saved_state**) in libFlipper-Folly.a(Benchmark.o)
  "boost::re_detail_106300::put_mem_block(void*)", referenced from:
      boost::re_detail_106300::perl_matcher<std::__1::__wrap_iter<char const*>, std::__1::allocator<boost::sub_match<std::__1::__wrap_iter<char const*> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::unwind_extra_block(bool) in libFlipper-Folly.a(Benchmark.o)
      boost::re_detail_106300::save_state_init::~save_state_init() in libFlipper-Folly.a(Benchmark.o)
  "boost::re_detail_106300::verify_options(unsigned int, boost::regex_constants::_match_flags)", referenced from:
      boost::re_detail_106300::perl_matcher<std::__1::__wrap_iter<char const*>, std::__1::allocator<boost::sub_match<std::__1::__wrap_iter<char const*> > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::find_imp() in libFlipper-Folly.a(Benchmark.o)
  "boost::re_detail_106300::raise_runtime_error(std::runtime_error const&)", referenced from:
      void boost::re_detail_106300::raise_error<boost::regex_traits_wrapper<boost::regex_traits<char, boost::cpp_regex_traits<char> > > >(boost::regex_traits_wrapper<boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::error_type) in libFlipper-Folly.a(Benchmark.o)
  "boost::re_detail_106300::get_default_error_string(boost::regex_constants::error_type)", referenced from:
      boost::re_detail_106300::cpp_regex_traits_implementation<char>::error_string(boost::regex_constants::error_type) const in libFlipper-Folly.a(Benchmark.o)
  "folly::init(int*, char***, bool)", referenced from:
      _main in libFlipper-RSocket.a(Benchmarks.o)
  "folly::detail::crc32_combine_hw(unsigned int, unsigned int, unsigned long)", referenced from:
      folly::crc32_combine(unsigned int, unsigned int, unsigned long) in libFlipper-Folly.a(Checksum.o)
  "folly::detail::crc32_combine_sw(unsigned int, unsigned int, unsigned long)", referenced from:
      folly::crc32_combine(unsigned int, unsigned int, unsigned long) in libFlipper-Folly.a(Checksum.o)
  "folly::detail::crc32c_combine_hw(unsigned int, unsigned int, unsigned long)", referenced from:
      folly::crc32c_combine(unsigned int, unsigned int, unsigned long) in libFlipper-Folly.a(Checksum.o)
  "folly::detail::crc32c_combine_sw(unsigned int, unsigned int, unsigned long)", referenced from:
      folly::crc32c_combine(unsigned int, unsigned int, unsigned long) in libFlipper-Folly.a(Checksum.o)
  "folly::detail::crc32c_hw(unsigned char const*, unsigned long, unsigned int)", referenced from:
      folly::crc32c(unsigned char const*, unsigned long, unsigned int) in libFlipper-Folly.a(Checksum.o)
  "yarpl::credits::add(long long, long long)", referenced from:
      yarpl::flowable::details::EmiterSubscription<long long>::request(long long) in libFlipper-RSocket.a(Flowables.o)
      yarpl::flowable::details::EmiterSubscription<rsocket::Payload>::request(long long) in libFlipper-RSocket.a(RequestResponseThroughputTcp.o)
      yarpl::flowable::details::EmiterSubscription<rsocket::Payload>::request(long long) in libFlipper-RSocket.a(StreamThroughputMemory.o)
      yarpl::flowable::details::EmiterSubscription<rsocket::Payload>::request(long long) in libFlipper-RSocket.a(StreamThroughputTcp.o)
  "yarpl::credits::consume(long long&, long long)", referenced from:
      yarpl::flowable::details::TrackingSubscriber<long long>::onNext(long long) in libFlipper-RSocket.a(Flowables.o)
      yarpl::flowable::details::TrackingSubscriber<rsocket::Payload>::onNext(rsocket::Payload) in libFlipper-RSocket.a(RequestResponseThroughputTcp.o)
      yarpl::flowable::details::TrackingSubscriber<rsocket::Payload>::onNext(rsocket::Payload) in libFlipper-RSocket.a(StreamThroughputMemory.o)
      yarpl::flowable::details::TrackingSubscriber<rsocket::Payload>::onNext(rsocket::Payload) in libFlipper-RSocket.a(StreamThroughputTcp.o)
  "boost::re_detail_106300::cpp_regex_traits_implementation<char>::transform_primary(char const*, char const*) const", referenced from:
      boost::cpp_regex_traits<char>::transform_primary(char const*, char const*) const in libFlipper-Folly.a(Benchmark.o)
  "boost::re_detail_106300::cpp_regex_traits_implementation<char>::transform(char const*, char const*) const", referenced from:
      boost::cpp_regex_traits<char>::transform(char const*, char const*) const in libFlipper-Folly.a(Benchmark.o)
ld: symbol(s) not found for architecture x86_64
`

**Comment by lorenc-tomasz** [comment]
It's strange as it's boost connected. Reverting to 4.2.x resolves the issue.

**Comment by lorenc-tomasz** [comment]
Example clean RN 0.63.4 example: https://1drv.ms/u/s!AuHh1Cwn9BGwuc1IxGurbtuQtyH8jQ?e=9X3Vfq

**Comment by lorenc-tomasz** [comment]
@countcain Could you look into this, please?

**Comment by lorenc-tomasz** [comment]
OK. I found the workaround. It's necessary to disable flipper which I would like not to disable in debug mode:

`  # Enables Flipper.
   #
   # Note that if you have use_frameworks! enabled, Flipper will not work and
   # you should disable these next few lines.
   #use_flipper!
   #post_install do |installer|
   #  flipper_post_install(installer)
   #end`

**Comment by countcain** [comment]
Hi @lorenc-tomasz I will take a look.

**Comment by lorenc-tomasz** [comment]
I thought that this is maybe an issue with cocoapods 1.9.3 but after updating to cocoapods 1.10.1 the issue persists.

**Comment by lorenc-tomasz** [comment]
@countcain Any idea how to resolve this?

**Comment by lorenc-tomasz** [comment]
Workaround I have found at the moment:
- user patch-packages
- use react-native-sentiance+4.3.1.patch:

```
diff --git a/node_modules/react-native-sentiance/.DS_Store b/node_modules/react-native-sentiance/.DS_Store
new file mode 100644
index 0000000..cb5113e
Binary files /dev/null and b/node_modules/react-native-sentiance/.DS_Store differ
diff --git a/node_modules/react-native-sentiance/ios/RNSentiance.podspec b/node_modules/react-native-sentiance/ios/RNSentiance.podspec
index d8657a2..228ca77 100644
--- a/node_modules/react-native-sentiance/ios/RNSentiance.podspec
+++ b/node_modules/react-native-sentiance/ios/RNSentiance.podspec
@@ -13,9 +13,7 @@ Pod::Spec.new do |s|
   s.source       = { :git => "https://github.com/sentiance/react-native-sentiance.git", :tag => "master" }
   s.source_files = "*.{h,m}"
   s.requires_arc = true
-  s.xcconfig     = { 'FRAMEWORK_SEARCH_PATHS' => '${PODS_ROOT}/SENTSDK' }
-
-  s.dependency "React"
-  s.dependency "SENTSDK", "5.9.0"
+  s.dependency "React-Core"
+  s.xcconfig = { 'FRAMEWORK_SEARCH_PATHS' => '${PODS_ROOT}/SENTSDK' }
 end
 
diff --git a/node_modules/react-native-sentiance/ios/SENTSDK.podspec b/node_modules/react-native-sentiance/ios/SENTSDK.podspec
new file mode 100644
index 0000000..f217432
--- /dev/null
+++ b/node_modules/react-native-sentiance/ios/SENTSDK.podspec
@@ -0,0 +1,17 @@
+Pod::Spec.new do |s|
+    s.name              = 'SENTSDK'
+    s.version           = '5.9.0'
+    s.summary           = 'The Sentiance iOS SDK.'
+    s.homepage          = 'https://sentiance.com/'
+
+    s.author            = { 'Name' => 'support@sentiance.com' }
+    s.license           = { :type => 'MIT' }
+    s.platform          = :ios
+    s.source            = { :http => 'https://sentiance-u1-sdk-downloads.s3-eu-west-1.amazonaws.com/ios/frameworks/SENTSDK-5.9.0.framework.zip' }
+    s.ios.deployment_target = '9.0'
+    s.frameworks = 'CoreMotion', 'SystemConfiguration', 'CoreLocation', 'Foundation', 'CallKit', 'CoreTelephony', 'CoreData'
+    s.libraries = 'z'
+    s.compiler_flags = '-lz', '-all_load', 'lc++'
+    s.resources = '**/SENTSDK.bundle'
+    s.vendored_frameworks = 'SENTSDK.framework', 'SENTSDK.framework/Frameworks/TensorFlowLiteC.framework'
+end

```

- preserve: pod 'SENTSDK', :podspec => '../node_modules/react-native-sentiance/ios/SENTSDK.podspec'
- pod install
- pod update SENTSDK

**Comment by netomi** [comment]
The problem is related to the '-all_load' flag present in the podspec of the SENTSDK pod. This was added in v5.9.0 of the SDK due to some bug in xcode at the time. This is not needed anymore and leads to weird side-effect when linking certain versions of boost in a react native environment. We have released 5.11.2 of the SENTSDK pod without this flag. This should resolve the issue.

Your workaround modifies the podspec to not have the -all_load flag as LD_FLAG but as compiler_flag which does not seem to trigger the problem.
